src/doLifeCycle.js

```diff
@@ -0,0 +1,84 @@
+class LifeCycle {
+
+  transform = value => value;
+  beforeUpdate = [];
+  afterUpdate = [];
+
+  constructor(main, nextValue, opts = {}) {
+    this.main = main;
+    this.nextValue = nextValue;
+
+    this.ensureArgs(opts);
+  }
+
+  ensureArgs = opts => {
+    if (typeof opts === 'function') {
+      this.transform = opts;
+    } else if (typeof opts === 'object') {
+      if (opts.beforeUpdate || opts.afterUpdate) {
+        if (opts.transform) this.transform = opts.transform;
+        this.beforeUpdate = this.ensureArray(opts.beforeUpdate);
+        this.afterUpdate = this.ensureArray(opts.afterUpdate);
+      }
+    } else throw new Error('Illegal options supplied to cycle.');
+  }
+
+  ensureArray = item => [].concat(item);
```

Comment:
this should be at the top as well :-)

it's squished between two bigger functions atm -- and serves more as a helper function and doesn't really have anything to do with the context of the other functions.